book_haven_api/views.py

```diff
@@ -24,12 +26,38 @@ def get_a_library(request, library_id):
             "detail": "The library with id " + str(library_id) + " does not exist."
         }]},status=status.HTTP_404_NOT_FOUND)
     
-@api_view(['GET'])
+@api_view(['GET', 'POST'])
 def get_library_books(request, library_id):
```

Comment:
Typically, you'd want to create two different methods. One for `GET` and one for `POST`. If you do want to keep them together, you'd probably want to switch to a DRF class `APIView` or something similar instead, which gives you the different HTTP methods by default. That way you can get rid of the branching logic in the method itself.

Reply:
We had that initially but ran into an error as it seemed like django hits the first url path that matches. Per what we had discussed with the front end, they had wanted it to be same "libraries/:id/books" end point for the post and get requests. Guess we will keep it for now but will revisit with the FE regarding that. Will also look into the APIView DRF class as we aren't super familiar with it yet

Reply:
Ah hm, maybe we have a Django add on package that parses it correctly, I'm honestly not sure. Regardless, anytime you are branching out into more than one HTTP verb, you'll probably want to use a DRF class, either `APIView` or one of the more [specialized classes like a `ListCreateApiView`](https://www.django-rest-framework.org/api-guide/generic-views/#listcreateapiview).